RingBuffer.fsx

```diff
@@ -7,15 +7,72 @@
 #load "RingBuffer.fs"
 open Hopac
 open Expecto
+open Hopac.Infixes
 
 let tests =
   testList "RingBuffer" [
     testCaseAsync "enqueue" (job {
-      let! rb = RingBuffer.create 2u
+      let! rb = RingBuffer.create 4us
+
+      let mutable result = []
+      
+      let takePerSecond = Job.delay <| fun () ->
+        timeOutMillis 100 ^=>. job {
+          let! res = RingBuffer.take rb
+          result <- ( res :: result)
+        }
+      Job.foreverServer takePerSecond |> run
+      
+      do! RingBuffer.put rb "a"
+      do! RingBuffer.put rb "b"
+      do! RingBuffer.put rb "c"
+      do! RingBuffer.put rb "d"
+      do! RingBuffer.put rb "e"
+      do! RingBuffer.put rb "f"
+      do! RingBuffer.put rb "g"
+
+      do! timeOutMillis 1000 ^=>. job {
```

Comment:
You could nicely use a `CountdownLatch` from Hopac to solve this problem, of waiting for 7 items.